# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ImageContainer.h
# Commit: 6fa7ede8ee41
# Full Hash: 6fa7ede8ee414dc872d5e5c5e68a5c71a63cc7a6
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2022-09-08 04:52:24
# Regressor Bug: 1781122
# File Overlap Count: 1
# Description:
#   Bug 1781122 Part 1: Make ChromiumCDMParent mark its Images as DRM. r=media-playback-reviewers,gfx-reviewers,alwu,sotaro
#   
#   This also modifies Image to track whether or not it is from a DRM source.
#   Later parts of the patch move this characteristic into the textures that
#   are used by compositors.
# ==============================================================================

diff -r a2c7bfa9fa8f -r 6fa7ede8ee41 gfx/layers/ImageContainer.h
--- a/gfx/layers/ImageContainer.h	Wed Sep 07 18:36:06 2022 +0000
+++ b/gfx/layers/ImageContainer.h	Wed Sep 07 18:51:11 2022 +0000
@@ -121,6 +121,9 @@
 
   int32_t GetSerial() const { return mSerial; }
 
+  bool IsDRM() const { return mIsDRM; }
+  void SetIsDRM(bool aIsDRM) { mIsDRM = aIsDRM; }
+
   virtual already_AddRefed<gfx::SourceSurface> GetAsSourceSurface() = 0;
 
   virtual bool IsValid() const { return true; }
@@ -155,7 +158,10 @@
       TextureClient* tcOverride = nullptr);
 
   Image(void* aImplData, ImageFormat aFormat)
-      : mImplData(aImplData), mSerial(++sSerialCounter), mFormat(aFormat) {}
+      : mImplData(aImplData),
+        mSerial(++sSerialCounter),
+        mFormat(aFormat),
+        mIsDRM(false) {}
 
   // Protected destructor, to discourage deletion outside of Release():
   virtual ~Image() = default;
@@ -168,6 +174,7 @@
   void* mImplData;
   int32_t mSerial;
   ImageFormat mFormat;
+  bool mIsDRM;
 
   static mozilla::Atomic<int32_t> sSerialCounter;
 };
